### PR TITLE
ABCU: auto pressurize newly built tiles

### PR DESCRIPTION
**Pull request by kbsmilk**
[GAME OBJECTS] [FEATURE]
## About the PR 
* Adds an option (defaulting to enabled) for the ABCU to pressurize its build area to standard station air.
* It has no material cost or downside.
* Moves APC building into the same code block for a minor benefit.

## Why's this needed? 
Pressurizing newly built stuff isn't very fun. This saves time, skipping to the fun part: using newly built stuff!

## Changelog 

```changelog
(u)Smilg
(+)ABCU can now automagically pressurize anything it builds.
```

### CODE REVIEW

**Comment by MeggalBozale** [comment]
Come on this should at least cost molitz for how much of a convenience it is

File: code/WorkInProgress/blueprints.dm [hunk 1]
```diff
@@ -286,9 +301,31 @@
...
-		src.apc_list = new/list
+		SPAWN(2 SECONDS) // gotta wait for make_tile() to finish
+			for (var/datum/objectinfo/N in src.apc_list)
+				new N.objecttype(src.apc_list[N])
+			src.apc_list = new/list
+
+			if (src.do_pressurize)
+				var/I
+				for (I = 1, I < length(src.turfs_to_pressurize), I++)
+					var/turf/simulated/T = src.turfs_to_pressurize[I]
```
Comment (TobleroneSwordfish) [inline code/WorkInProgress/blueprints.dm:312]:
```suggestion
				for (var/turf/simulated/T as anything in src.turfs_to_pressurize)
```

File: code/WorkInProgress/blueprints.dm [hunk 2]
```diff
@@ -286,9 +301,31 @@
...
+						gas_add.nitrogen = clamp(MOLES_N2STANDARD * count - T.parent.air.nitrogen, 0, INFINITY)
+						T.parent.air.merge(gas_add)
+						src.turfs_to_pressurize.Remove(T.parent.members)
+					else
+						var/datum/gas_mixture/gas_add = new /datum/gas_mixture
+						gas_add.temperature = T20C
+						gas_add.oxygen = clamp(MOLES_O2STANDARD - T.air.oxygen, 0, INFINITY)
+						gas_add.nitrogen = clamp(MOLES_N2STANDARD - T.air.nitrogen, 0, INFINITY)
+						T.assume_air(gas_add)
+			src.turfs_to_pressurize = new/list
```
Comment (TobleroneSwordfish) [inline code/WorkInProgress/blueprints.dm:328]:
```suggestion
			src.turfs_to_pressurize = list()
```

File: code/WorkInProgress/blueprints.dm [hunk 3]
```diff
@@ -286,9 +301,31 @@
...
+						gas_add.oxygen = clamp(MOLES_O2STANDARD * count - T.parent.air.oxygen, 0, INFINITY)
+						gas_add.nitrogen = clamp(MOLES_N2STANDARD * count - T.parent.air.nitrogen, 0, INFINITY)
+						T.parent.air.merge(gas_add)
+						src.turfs_to_pressurize.Remove(T.parent.members)
+					else
+						var/datum/gas_mixture/gas_add = new /datum/gas_mixture
+						gas_add.temperature = T20C
+						gas_add.oxygen = clamp(MOLES_O2STANDARD - T.air.oxygen, 0, INFINITY)
+						gas_add.nitrogen = clamp(MOLES_N2STANDARD - T.air.nitrogen, 0, INFINITY)
+						T.assume_air(gas_add)
```
Comment (TobleroneSwordfish) [inline code/WorkInProgress/blueprints.dm:327]: `assume_air` handles group processing internally, you don't need to do this

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
should probably either cost molitz or (cooler) require a connected canister that it uses to pressurize

**Comment by kbsmilk** [comment]
This is going to take a Lot more code for material consumption

**Pull request closed without merging**